contracts/btc-finality/src/contract.rs

```diff
@@ -228,10 +233,14 @@ fn handle_update_staking(
 }
 
 fn handle_begin_block(deps: &mut DepsMut, env: Env) -> Result<Response, ContractError> {
+    // Distribute rewards to finality providers
+    distribute_rewards_fps(deps, &env)?;
```

Comment:
```suggestion
    // Distribute rewards to finality providers
    record_fp_rewards(deps, &env)?;
```

The function is only doing accounting but not actual reward distribution